src/library_html5.js

```diff
@@ -28,42 +29,41 @@ var LibraryHTML5 = {
     visibilityChangeEvent: 0,
     touchEvent: 0,
 
-    // When we transition from fullscreen to windowed mode, we remember here the element that was just in fullscreen mode
-    // so that we can report information about that element in the event message.
+    // When we transition from fullscreen to windowed mode, we remember here the
+    // element that was just in fullscreen mode so that we can report
+    // information about that element in the event message.
     previousFullscreenElement: null,
 
 #if MIN_SAFARI_VERSION <= 80000 || MIN_CHROME_VERSION <= 21 // https://caniuse.com/#search=movementX
-    // Remember the current mouse coordinates in case we need to emulate movementXY generation for browsers that don't support it.
+    // Remember the current mouse coordinates in case we need to emulate
+    // movementXY generation for browsers that don't support it.
     // Some browsers (e.g. Safari 6.0.5) only give movementXY when Pointerlock is active.
     previousScreenX: null,
     previousScreenY: null,
 #endif
 
-    // When the C runtime exits via exit(), we unregister all event handlers added by this library to be nice and clean.
+    // When the C runtime exits via exit(), we unregister all event handlers
+    // added by this library to be nice and clean.
     // Track in this field whether we have yet registered that __ATEXIT__ handler.
     removeEventListenersRegistered: false,
 
 #if HTML5_SUPPORT_DEFERRING_USER_SENSITIVE_REQUESTS
-    // If we are in an event handler, specifies the event handler object from the eventHandlers array that is currently running.
+    // If we are in an event handler, specifies the event handler object from
+    // the eventHandlers array that is currently running.
     currentEventHandler: null,
 #endif
 */
 
-    // If positive, we are currently executing in a JS event handler.
-    // (this particular property must be initialized to zero, as we ++/-- it)
-    inEventHandler: 0,
-
     removeAllEventListeners() {
-      for (var i = JSEvents.eventHandlers.length-1; i >= 0; --i) {
-        JSEvents._removeHandler(i);
+      while (JSEvents.eventHandlers.length) {
+        JSEvents._removeHandler(0);
```

Comment:
This seems to make the behavior quadratic? Removing from the end is more efficient than removing from the start, in a flat implementation of a vector, which this array should be.

Reply:
This code only runs exactly once though.. and the number of handlers here is likely tiny (5-10 max).  So I don't think the perf here matters at all.

Reply:
I agree it's likely little work, but the worst case can be annoying, even if it is unlikely. Is it too costly to do this do you think?
```suggestion
        JSEvents._removeHandler(JSEvents.eventHandlers.length - 1);
```

Reply:
OK, Done